### PR TITLE
Use programmatic API

### PR DESCRIPTION
**Pull request by bored**
Use the programmatic API as said in https://github.com/sindresorhus/ava/issues/345.

### CODE REVIEW

**Comment by paulohp** [comment]
@sindresorhus do you have checked this ?

File: index.js [hunk 1]
```diff
@@ -22,15 +27,43 @@ module.exports = function () {
 
 		cb(null, file);
 	}, function () {
-		files.unshift(BIN);
+		options = extend({
```
Comment (SamVerschueren) [inline index.js]: Use `object-assign` for this

**Comment by sindresorhus** [comment]
Sorry about the delay @bored @paulohp. I'm just not happy about how much boilerplate is required to use the AVA API. Feels like we have to reimplement almost the whole AVA `cli.js`.

@vdemedes I think we either need to improve the API or expose some kind of `CliApi`, of which is what's in this PR. // @jamestalmage @novemberborn

**Comment by jamestalmage** [comment]
I think I have a start on the CLIApi. I was working on extracting the config - and it kinda turned into that.

**Comment by Shiggiddie** [comment]
@jamestalmage and/or @sindresorhus after looking through `gulp-ava` and `ava` repos, it looks like this is still an outstanding issue.

An inability to dynamically generate and pass config into `gulp-ava` is likely to block the adoption of `ava` for a current work project, despite my desire to harness the framework.  Is there a clear direction on programmatic API in `gulp-ava` planned at this time?

**Comment by novemberborn** [comment]
I've opened https://github.com/avajs/ava/issues/1043 to discuss the necessary API work in AVA itself.

**Comment by sindresorhus** [comment]
Closing this for now. We can reopen when the AVA API is stable.

**Pull request closed without merging**